### DISCUSSION TITLE
Missing buttons on devices

### DISCUSSION THREAD

**Discussion by rusomenace** [netbox-community/netbox#11924] [Help Wanted!] [answered]
Hi, a few days ago we implement netbox, during installation everything went fine no errors.
On use we see something extrange under devices, the button for single items in the "device components" section does not exist, the only option is "import"
Coding devices its hard two because there is not much examples, see the attach image

![image](https://user-images.githubusercontent.com/67525460/223471957-6e9ad349-a0ba-49c4-9534-a56b11d588f4.png)

system: ubuntu 22.04 fully updated
netbox version:  v3.4.5

Thanks

**Comment by candlerb** [comment, marked as answer]
Yes, that's correct. Device components cannot exist in isolation, only when part of a device.  So you need to navigate first to the device where you want to add them, then you'll see an "Add Components..." dropdown.

![image](https://user-images.githubusercontent.com/44789/223479003-1418d32a-b747-44c0-9834-6e6f11be033f.png)

> Coding devices its hard two because there is not much examples, see the attach image

I don't understand the question: by "coding" do you mean "creating", or are you actually talking about software programming?

In order to create a Device, you'll first need to create a Manufacturer, a Device Type, and a Site. Then you can add a Device of that type, in that site.  (You may also need to add a Device Role: I can't remember whether a freshly-installed Netbox comes with a default set of Roles).

**Reply by rusomenace** [reply]
Thanks for the clarification it was my mistake of understanding on how to add items.
The "Coding" i was meaning its related to build the bulk import as CSV, JSON or YAML format. Its a wrong way we use here to express ourself because of the IDE VS o VS Code that we generally use to create yaml o json files.

Thanks
